CSP_Minizinc_Code.py

```diff
@@ -0,0 +1,179 @@
+        self.__repo_image=repo_image
+
+    def get_robot_status(self):
+        #Robot senses the current state of its environment
+        #Initially the sensed data is provided explicitly
+        self.__gripper_status=True
+        self.__robot_in_motion=False
+        self.__finger_open=True
+
+    def update_info_to_repo(self):
+        #Updates the current state of the robot to the repository
+        self.__repo_image.update_context(self.__gripper_status,self.__robot_in_motion,self.__finger_open)
+
+
+#Repository contains information for deployment. It is the central component in the architecture. Every components updates the information to the repository. 
+class Repository():
+    def __init__(self, active_safety_monitor = SafetyMonitor.NO_SELECTION, active_sensor = Sensors.NO_SELECTION,
+                 current_context = [False, False, False],fixed_deployment = True,
+                 safety_monitor_platform = Platforms.NO_SELECTION):
+
+        self.__active_safety_monitor = active_safety_monitor
+        self.__active_sensor = active_sensor
+        self.__safety_monitor_platform = safety_monitor_platform
+        self.__current_context = current_context
+        self.__fixed_deployment = fixed_deployment
+
+    #Setter methods for updating current context, safety monitor and platform information for deployment from other componenents
+    def update_context(self,gripper_status:bool,robot_motion:bool,finger_open=bool):
+        self.__current_context = gripper_status,robot_motion,finger_open
+
+    def update_current_safety_monitor(self,current_safety_monitor):
+        self.__active_safety_monitor = current_safety_monitor
+        print("Selected safety monitor:",self.__active_safety_monitor)
+
+    def update_platform_status(self,safety_monitor_platform):
+        self.__safety_monitor_platform = safety_monitor_platform
+        print("\nSelected platform to deploy {0}: PF{1} ".format(self.__active_safety_monitor,self.__safety_monitor_platform))
+
+    #Getter methods for obtaining current context and safety monitor information from other componenents
+    def get_active_safety_monitor(self):
+        return self.__active_safety_monitor
+
+    def get_current_context(self):
+        return self.__current_context
+
+    def __notify_changes(self):
+        pass
+
+
+#Interface for querying and updating repository and acts as a communicator between repository and other components of the system 
+class Selector(ABC):
+    @abstractmethod
+    def query_repository(self):
+        pass
+
+    @abstractmethod
+    def update_repository(self):
+        pass
+
+#Contains definition for selecting the safety monitor based on the current context. The selected safety monitor information is updated in the repository.
+class Safety_Monitor_Selector(Selector):
+    def __init__(self,repo_image):
+        #Initially the attributes are set to null and no safety monitor is selected by default.
+        self.__current_context = None

```

Comment:
This is a list of three elements, not one. We have to come up with a convention to intialize bools. I would prefer everything to be intialized to False. Give your opinion